--- gst-plugins-base.eclass.ori	2010-11-13 15:42:28.000000000 +0100
+++ /usr/portage/eclass/gst-plugins-base.eclass	2010-11-13 15:40:05.000000000 +0100
@@ -94,22 +94,22 @@
 
 	# Link with the syswide installed gst-libs if needed
 	gst-plugins10_find_plugin_dir
-	sed -e "s:\$(top_builddir)/gst-libs/gst/interfaces/libgstinterfaces:${ROOT}/usr/$(get_libdir)/libgstinterfaces:" \
-		-e "s:\${top_builddir}/gst-libs/gst/interfaces/libgstinterfaces:${ROOT}/usr/$(get_libdir)/libgstinterfaces:" \
-		-e "s:\$(top_builddir)/gst-libs/gst/audio/libgstaudio:${ROOT}/usr/$(get_libdir)/libgstaudio:" \
-		-e "s:\${top_builddir}/gst-libs/gst/audio/libgstaudio:${ROOT}/usr/$(get_libdir)/libgstaudio:" \
-		-e "s:\$(top_builddir)/gst-libs/gst/cdda/libgstcdda:${ROOT}/usr/$(get_libdir)/libgstcdda:" \
-		-e "s:\${top_builddir}/gst-libs/gst/cdda/libgstcdda:${ROOT}/usr/$(get_libdir)/libgstcdda:" \
-		-e "s:\$(top_builddir)/gst-libs/gst/riff/libgstriff:${ROOT}/usr/$(get_libdir)/libgstriff:" \
-		-e "s:\${top_builddir}/gst-libs/gst/riff/libgstriff:${ROOT}/usr/$(get_libdir)/libgstriff:" \
-		-e "s:\$(top_builddir)/gst-libs/gst/tag/libgsttag:${ROOT}/usr/$(get_libdir)/libgsttag:" \
-		-e "s:\${top_builddir}/gst-libs/gst/tag/libgsttag:${ROOT}/usr/$(get_libdir)/libgsttag:" \
-		-e "s:\$(top_builddir)/gst-libs/gst/video/libgstvideo:${ROOT}/usr/$(get_libdir)/libgstvideo:" \
-		-e "s:\${top_builddir}/gst-libs/gst/video/libgstvideo:${ROOT}/usr/$(get_libdir)/libgstvideo:" \
-		-e "s:\$(top_builddir)/gst-libs/gst/netbuffer/libgstnetbuffer:${ROOT}/usr/$(get_libdir)/libgstnetbuffer:" \
-		-e "s:\${top_builddir}/gst-libs/gst/netbuffer/libgstnetbuffer:${ROOT}/usr/$(get_libdir)/libgstnetbuffer:" \
-		-e "s:\$(top_builddir)/gst-libs/gst/rtp/libgstrtp:${ROOT}/usr/$(get_libdir)/libgstrtp:" \
-		-e "s:\${top_builddir}/gst-libs/gst/rtp/libgstrtp:${ROOT}/usr/$(get_libdir)/libgstrtp:" \
+	sed     -e "s:\$[({]top_builddir[)}]/gst-libs/gst/interfaces/libgstinterfaces-\$(GST_MAJORMINOR).la:-lgstinterfaces-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/interfaces/libgstinterfaces-@GST_MAJORMINOR@.la:-lgstinterfaces-@GST_MAJORMINOR@:" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/audio/libgstaudio-\$(GST_MAJORMINOR).la:-lgstaudio-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/audio/libgstaudio-@GST_MAJORMINOR@.la:-lgstaudio-@GST_MAJORMINOR@:" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/cdda/libgstcdda-\$(GST_MAJORMINOR).la:-lgstcdda-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/cdda/libgstcdda-@GST_MAJORMINOR@.la:-lgstcdda-@GST_MAJORMINOR@:" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/riff/libgstriff-\$(GST_MAJORMINOR).la:-lgstriff-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/riff/libgstriff-@GST_MAJORMINOR@.la:-lgstriff-@GST_MAJORMINOR@:" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/tag/libgsttag-\$(GST_MAJORMINOR).la:-lgsttag-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/tag/libgsttag-@GST_MAJORMINOR@.la:-lgsttag-@GST_MAJORMINOR@:" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/video/libgstvideo-\$(GST_MAJORMINOR).la:-lgstvideo-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/video/libgstvideo-@GST_MAJORMINOR@.la:-lgstvideo-@GST_MAJORMINOR@:" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/netbuffer/libgstnetbuffer-\$(GST_MAJORMINOR).la:-lgstnetbuffer-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/netbuffer/libgstnetbuffer-@GST_MAJORMINOR@.la:-lgstnerbuffer-@GST_MAJORMINOR@:" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/rtp/libgstrtp-\$(GST_MAJORMINOR).la:-lgstrtp-\$(GST_MAJORMINOR):" \
+		-e "s:\$[({]top_builddir[)}]/gst-libs/gst/rtp/libgstrtp-@GST_MAJORMINOR@.la:-lgstrtp-@GST_MAJORMINOR@:" \
 		-i Makefile.in
 #	cd ${S}
 
